### PR TITLE
Fix query for NodeExecutionsRepo.Count

### PR DESCRIPTION
**Pull request by andrewwdye**
Signed-off-by: Andrew Dye <andrewwdye@gmail.com>

# TL;DR
Recently added query for `NodeExecutionsRepo.Count` is broken. This change updates the query to be consistent with `NodeExecutionsRepo.List`.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
The `NodeExecutionsRepo.Count` borrowed join conditions from `NodeExecutionsRepo.ListEvents`, which is currently broken. This change fixes the query to match `NodeExecutionsRepo.List`. I updated the unittest and validated the change with a local instance of flyteadmin and postgres.

## Tracking Issue
NA

## Follow-up issue
Opened https://github.com/flyteorg/flyte/issues/2858 for the broken `ListEvents` implementation.

### CODE REVIEW

**Review by katrogan — COMMENTED** [review]
would you also mind cleaning up references to ListEvents since it appears to be unused? https://sourcegraph.com/github.com/flyteorg/flyteadmin@master/-/blob/pkg/repositories/interfaces/node_execution_repo.go?L24:2#tab=references

**Comment by katrogan** [comment]
ah, I see you opened up the issue to track - never mind. thanks!

**Comment by andrewwdye** [comment]
> would you also mind cleaning up references to ListEvents since it appears to be unused? https://sourcegraph.com/github.com/flyteorg/flyteadmin@master/-/blob/pkg/repositories/interfaces/node_execution_repo.go?L24:2#tab=references

Yep, https://github.com/flyteorg/flyteadmin/pull/473

**Review by katrogan — APPROVED** [review]
